app/provider/app.go

```diff
@@ -374,15 +373,6 @@ func New(
 		authcodec.NewBech32Codec(sdk.GetConfig().GetBech32ValidatorAddrPrefix()),
 		authcodec.NewBech32Codec(sdk.GetConfig().GetBech32ConsensusAddrPrefix()),
 	)
-	app.MintKeeper = mintkeeper.NewKeeper(
-		appCodec,
-		runtime.NewKVStoreService(keys[minttypes.StoreKey]),
-		app.StakingKeeper,
-		app.AccountKeeper,
-		app.BankKeeper,
-		authtypes.FeeCollectorName,
-		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
-	)
```

Comment:
`DistrKeeper` should use the staking keeper or the provider one?

Reply:
See https://informalsystems.notion.site/Staking-dependencies-in-Cosmos-SDK-v0-50-5-and-ibc-go-v8-3-2-669889523e1840f68996f69946bf388e?pvs=4
They are all ok using staking

Reply:
Oh, yes. Thanks. Is there a problem with using the `provider` keeper and hence the consensus-active validator set even though they're fine using the `staking` keeper? Is there a potentiality where distr, slashing, evidence, ... might need to use non-consensus-active validators?

Reply:
It just means our provider keeper would need to implement the interface that these keepers expect from the staking keeper, too, so it just increases the surface of the provider keeper for no benefit